util/state.js

```diff
@@ -187,23 +187,29 @@ return 0;
   }
 
   async setStartTime() {
-    const startTime = this._timestamp();
-    return await this.redis.rpush(`${this.startkey}:${this.uid}`, startTime);
-  }
-
-  async getStartTimes() {
-    return await this.redis.lrange(`${this.startkey}:${this.uid}`, 0, -1);
+    const startTimeTs = Date.now();
+    await this.incExecTime(startTimeTs);
```

Comment:
```suggestion
    // if previous start time exists, incr execution time by difference before setting new time
    await this.incExecTime(startTimeTs);
```
Something like that? Otherwise from the code itself without context it's not super obvious what's going on here